crates/merod/src/cli/config.rs

```diff
@@ -1,209 +1,76 @@
 #![allow(unused_results, reason = "Occurs in macro")]
 
-use core::net::IpAddr;
+use std::env::temp_dir;
 use std::fs::{read_to_string, write};
+use std::str::FromStr;
 
-use calimero_network::config::BootstrapNodes;
-use clap::{Args, Parser, ValueEnum};
-use eyre::{eyre, Result as EyreResult};
-use multiaddr::{Multiaddr, Protocol};
-use toml_edit::{DocumentMut, Value};
+use calimero_config::{ConfigFile, CONFIG_FILE};
+use camino::Utf8PathBuf;
+use clap::{value_parser, Parser};
+use eyre::{bail, eyre, Result as EyreResult};
+use toml_edit::{Item, Value};
 use tracing::info;
 
 use crate::cli;
 
-#[derive(Copy, Clone, Debug, ValueEnum)]
-pub enum ConfigProtocol {
-    Near,
-    Starknet,
-}
-
-impl ConfigProtocol {
-    pub fn as_str(&self) -> &str {
-        match self {
-            ConfigProtocol::Near => "near",
-            ConfigProtocol::Starknet => "starknet",
-        }
-    }
-}
-
 /// Configure the node
 #[derive(Debug, Parser)]
 pub struct ConfigCommand {
-    /// List of bootstrap nodes
-    #[arg(long, value_name = "ADDR", conflicts_with = "boot_network")]
-    pub boot_nodes: Vec<Multiaddr>,
-
-    /// Use nodes from a known network
-    #[arg(long, value_name = "NETWORK", conflicts_with = "boot_nodes")]
-    pub boot_network: Option<BootstrapNetwork>,
-
-    /// Host to listen on
-    #[arg(long, value_name = "HOST", use_value_delimiter = true)]
-    pub swarm_host: Vec<IpAddr>,
-
-    /// Port to listen on
-    #[arg(long, value_name = "PORT")]
-    pub swarm_port: Option<u16>,
-
-    /// Host to listen on for RPC
-    #[arg(long, value_name = "HOST", use_value_delimiter = true)]
-    pub server_host: Vec<IpAddr>,
-
-    /// Port to listen on for RPC
-    #[arg(long, value_name = "PORT")]
-    pub server_port: Option<u16>,
-
-    #[command(flatten)]
-    pub mdns: Option<MdnsArgs>,
+    /// Key-value pairs to be added or updated in the TOML file
+    #[clap(short, long, value_parser = value_parser!(KeyValuePair))]
```

Comment:
```suggestion
```

You should be able to skip `value_parser` as `T: FromStr` is implied. Also, can we skip the `--arg` flag?